forgot.cgi

```diff
@@ -26,9 +26,33 @@ time() - $link{'time'} > 60*$timeout &&
 # Get the Webmin user
 &foreign_require("acl");
 my ($wuser) = grep { $_->{'name'} eq $link{'user'} } &acl::list_users();
-$wuser || &error(&text('forgot_euser2',
+
+# Get the Virtualmin mail Unix user if Webmin user is not found
+my ($muser, $muserdom);
+if (!$wuser && $link{'muser'}) {
+	# Probably Virtualmin mail user, so try to find it
+	&foreign_require("virtual-server");
+	foreach my $d (&virtual_server::list_domains()) {
```

Comment:
This function should speed up finding the domain for a random username : https://github.com/virtualmin/virtualmin-gpl/blob/master/virtual-server-lib-funcs.pl#L12516

Reply:
Thanks for the heads-up! If there’s nothing else, could you go ahead and merge it and fix it? Otherwise, I’ll have to wait until tomorrow morning to handle it.

Reply:
> This function should speed up finding the domain for a random username : https://github.com/virtualmin/virtualmin-gpl/blob/master/virtual-server-lib-funcs.pl#L12516

Done! Thanks! Check this patch out https://github.com/webmin/webmin/commit/2d5768a6123fea5b7fb0ffa84f2e9aeabe721977.